contents/tutorials/react-charts.md

```diff
@@ -0,0 +1,454 @@
+---
+title: How to use React Charts to visualize analytics data (with examples)
+date: 2024-08-23
+author:
+ - ian-vanagas
+tags:
+ - product analytics
+---
+
+React Charts is a popular visualization and charting library for React. It provides a simplified set of performant charts to use with analytics data from PostHog.
+
+To provide examples of what you can do with it, we create a basic Next.js app, set it up to get data from PostHog's query API, and visualize it with React Charts.
+
+> **Don't want to do all the work of querying and visualizing?** You can always share or embed insights, dashboards, and more. See [sharing docs](/docs/product-analytics/sharing) for more.
+
+## Creating a Next.js app
+
+Because we need to query private data, we need a way to make server side requests. Next.js makes this easy (which is why we chose it).
```

Comment:
I dont quite follow why this is?

Reply:
your PostHog data is private, querying it from the client side exposes your personal API key that could be used to query whatever data from PostHog you wanted.

Reply:
Yes, I put this warning below

> > **Note:** You'll likely want to include the payload on the server-side too so users can't make requests to this endpoint with arbitrary queries. We don't do this in this tutorial to keep the changes in the examples below in a single file.

Reply:
I mean something more like 
> `!!!` **WARNING** `!!!` **This will expose all data in your PostHog account to the public internet** `!!!` **WARNING** `!!!` YOU DO NOT WANT TO DO THIS IN REAL LIFE

Reply:
🫡

Reply:
And then a followup tutorial: how to do this securely 😅